jlm/hls/backend/rvsdg2rhls/UnusedStateRemoval.cpp

```diff
@@ -14,7 +14,7 @@ namespace jlm::hls
 {
 
 static bool
-IsPassthroughArgument(const rvsdg::RegionArgument & argument)
+IsPassthroughArgument(const rvsdg::output & argument)
```

Comment:
Why did you make it more general? I understand that this most likely comes from the decision that you made the argument in the ContextVar and rvsdg::output, but why? We should always give the most specific type back, no?

Reply:
yes the intent is to get rid of RegionResult (it doesn't really contribute anything besides being a redundant type)